content/blog/elevate-your-v-project-with-unit-tests.md

```diff
@@ -0,0 +1,749 @@
+		},
+		RunConfig{
+			shape: 'left-triangle'
+			output: ['*', '**', '***', '****', '*****']
+		},
+	]
+)
+
+fn test_all_options() {
+	for case in all_options_test_cases {
+		result := os.execute_or_panic('${os.quoted_path(@VEXE)} run . --shape $case.shape --size $case.size --symbol "$case.symbol"')
+
+		assert result.exit_code == 0
+		assert result.output.split_into_lines() == case.output
+	}
+}
+
+fn test_shapes_only() {
+	for case in shape_only_test_cases {
+		result := os.execute_or_panic('${os.quoted_path(@VEXE)} run . --shape $case.shape')
+
+		assert result.exit_code == 0
+		assert result.output.split_into_lines() == case.output
+	}
+}
+
+```
+
+As always, we run the tests with `v test .` from the root of the project.
+If we need to run only the unit tests, we can type ` v test test/unit_tests`
+instead. Remember to use `Tab` for autocomplete support for paths.
+
+## BONUS: Running Specific Tests
+
+If you run `v help test`, you will come across a lot of useful information.
+Most of it has been demonstrated in this article, but there are some tid-bits
+that you may find intriguing. One of these is the `-run-only` option:
+
+```text
+If you give `-run-only GPATTERN`, then *only* test functions, that do
+match by name the given glob pattern `GPATTERN` will run. You can separate
+multiple glob patterns with `,`.
+If a _test.v file lacks matching functions for all of the glob patterns, it
+will be ignored completely, so you can do in effect:
+    `v test . -run-only test_your_fn_name`
+... and V will run only that test function, no matter how many _test.v
+files you have, and how many other test_ functions exist in them.
+NB: glob patterns support `*` which matches anything, and `?`, that
+matches any single character. They are *NOT* regular expressions however.
+```
+
+Let's try it on our project!
+
+```text
+$ v test . -run-only test_all_options
+---- Testing... ------------------------------------------------------------------------
+ OK     221.005 ms /path/to/geo/test/command_line_tests/multi_option_test.v
+----------------------------------------------------------------------------------------
+Summary for all V _test.v files: 1 passed, 1 total. Runtime: 221 ms, on 1 job.
+
+$ v test . -run-only *generate*
+---- Testing... ------------------------------------------------------------------------
+ OK    [1/2]   231.238 ms /path/to/geo/test/unit_tests/quadrilateral_test.v
+ OK    [2/2]   240.615 ms /path/to/geo/test/unit_tests/triangle_test.v
+----------------------------------------------------------------------------------------
+Summary for all V _test.v files: 2 passed, 2 total. Runtime: 241 ms, on 2 parallel jobs.
+```
+
+## Wrapping Up
+
+This was a detailed example of how one would go about adding unit tests
+to their V projects. Some teams may practice **Test Driven Development**
+where the unit tests are written before the actual implementation to
+ensure that the API design goal is met. In any case, tests are good to
```

Comment:
the sentence about Test Driven Development can be deleted imho

Reply:
then the next sentence can become just:
`Tests are good to have and instill confidence in the maintainers of a project.`
The users of a project do not care at all usually how the sausage is made.